### ISSUE TITLE
allow sampling  to a fixed number of reads

### ISSUE THREAD

**Issue by bwlang** [samtools/samtools#955]
#### Is your feature request related to a problem? Please specify.
When downsampling a bunch of bam files, one has to calculate fractions of reads  to retain for each bam.  Which can be a hassle.
#### Describe the solution you would like.

Perhaps a reservoir sampling option to avoid a double walk of the sam?
Maybe would be good to support sampling on a per read group basis too.

**Comment by tseemann** [comment]
I agree it would be good to have `N` as well as `fraction`, just like `seqtk sample` does.

The code for the reservoir sampling in `seqtk` is here:
https://github.com/lh3/seqtk/blob/master/seqtk.c#L1025-L1132

**Comment by daviesrob** [comment]
See previous discussion [here](https://github.com/samtools/samtools/issues/931#issuecomment-416975684) including how you can do what you want (albeit not very efficiently) using `samtools collate`.

If we decide to support this, it'll likely need a dedicated command.  It's likely to be a bit too complicated to squeeze it into `samtools view`.
